File: Data Structures & Algorithms/buy-and-sell-crypto-with-cooldown/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        return dfs(prices, true, 0);
    }

    int dfs(vector<int>& prices, bool buy, int i) {
        if (i >= prices.size()) return 0;
        int first = dfs(prices, buy, i + 1);
        if (buy) {
            return max(first, dfs(prices, false, i + 1) - prices[i]);
        } else {
            return max(first, dfs(prices, true, i + 2) + prices[i]);
        }
    }
};

File: Data Structures & Algorithms/max-area-of-island/submission-1.java
class Solution {
    public int maxAreaOfIsland(int[][] grid) {
        int rows = grid.length;
        int cols = grid[0].length;
        int max = 0;
        for (int i = 0; i < grid.length; i++) {
            for (int j = 0; j < grid[i].length; j++) {
                max = Math.max(dfs(grid, new boolean[rows][cols], i, j), max);
            }
        }
        return max;
    }

    public int dfs(int[][] grid, boolean[][] visited, int r, int c) {
        if (r < 0 || r >= grid.length || c < 0 || c >= grid[0].length) return 0;
        if (grid[r][c] == 0) return 0;
        if (visited[r][c]) return 0;
        visited[r][c] = true;
        return dfs(grid, visited, r + 1, c) + dfs(grid, visited, r - 1, c) + dfs(grid, visited, r, c + 1) + dfs(grid, visited, r, c - 1) + 1;
    }
}
